frontend/e2e/signup.spec.js

```diff
@@ -0,0 +1,80 @@
+const { test, expect } = require('@playwright/test');
+
+/**
+ * Basic End-to-End Test for Stockify Signup
+ * Prerequisites:
+ * - Backend server running on http://127.0.0.1:5001
+ * - Frontend server running (port will be determined automatically)
+ * 
+ * Test Steps:
+ * 1. Verify servers are running
+ * 2. Navigate to signup page
+ * 3. Fill in form
+ * 4. Submit form
+ * 5. Verify success
+ */
+
+test.describe('Basic Signup Flow', () => {
+  let frontendPort;
+
+  test.beforeAll(async ({ request }) => {
+    // Check if backend is running
+    try {
+      const response = await request.get('http://127.0.0.1:5001/');
+      expect(response.ok()).toBeTruthy();
+    } catch (error) {
+      throw new Error('Backend server is not running. Please start it with `flask run --port=5001`');
+    }
+
+    // Try to determine the frontend port by checking common Next.js ports
+    for (const port of [3000, 3001, 3002, 3003]) {
```

Comment:
Hmm... typically the port for the application is configured in the application configuration and is known ahead of time. I'm curious why you chose this approach?